[DeepSeekV3.2] Add pure TP+MTP test

ashtonchew opened the pull request:
## Motivation

Add test coverage for DeepSeek V3.2 with pure tensor parallel mode and MTP (EAGLE speculative decoding) without DP attention.

This addresses the "Support pure TP+MTP" item from the DeepSeek V3.2 optimization roadmap (#15025).

## Modifications

- Add `test/srt/test_deepseek_v32_tp_mtp.py` - test for pure TP+MTP configuration
- Register test in `test/srt/run_suite.py` for `per-commit-8-gpu-h200` suite  
- Update `docs/basic_usage/deepseek_v32.md` with Pure TP + MTP example

## Accuracy Tests

Will be validated by CI (`per-commit-8-gpu-h200` suite):
- GSM8K accuracy > 0.935
- Speculative acceptance length > 2.5

## Benchmarking and Profiling

Will be validated by CI:
- Speed > 70 tok/s on batch size 1

ashtonchew committed:
feat(deepseek-v32): add pure TP+MTP test and documentation

Add test coverage for DeepSeek V3.2 with pure tensor parallel mode
and MTP (EAGLE speculative decoding) without DP attention.

Fridge003 commented on test/srt/test_deepseek_v32_tp_mtp.py:
```diff
@@ -0,0 +1,112 @@
+"""Test DeepSeek V3.2 with pure TP (no DP attention) and MTP (EAGLE speculative decoding).
```
Can we make it a subtest of 
`test/nightly/test_deepseek_v32_tp.py`

It's better to make this test a nightly one

Fridge003 commented:
Also can you post some performance data of tp8 vs tp8+mtp?

ashtonchew commented:
> Also can you post some performance data of tp8 vs tp8+mtp?

Will do later today, I will first push the changes of making it a subtest to make it a nightly run

ashtonchew committed:
refactor(test): move TP+MTP test to nightly suite

Move DeepSeek V3.2 TP+MTP test to nightly as a third variant in
test_deepseek_v32_tp.py. Update summary table to include MTP metrics.

ashtonchew commented:
I added the changes you requested @Fridge003 and I don't have access to H200 GPUs to run the benchmark. The nightly CI on H200 will produce these numbers once the test runs. Would you be able to run a quick benchmark on your end, or should we wait for CI?

Fridge003 commented:
I can check this later today

Fridge003 commented on test/nightly/test_deepseek_v32_tp.py:
```diff
@@ -134,10 +137,99 @@ def test_a_gsm8k(
+            TEST_RESULTS.append(
+                {
+                    "variant": "tp_mtp",
+                    "prefill_backend": "default",
```
prefill/decode backend should be `flashmla_sparse` and `flashmla_kv`

ashtonchew committed:
fix: use flashmla_sparse and flashmla_kv for TP+MTP test results

Fridge003 committed:
Update test/nightly/test_deepseek_v32_tp.py
